### ISSUE TITLE
[a11y]: The scrollable element <pre> with non-interactive content is not tabbable

### ISSUE THREAD

**Issue by zhanttbj** [carbon-design-system/carbon#14693]
### Package

carbon-components

### Browser

Chrome

### Operating System

Windows

### Package version

V10

### Automated testing tool and ruleset

IBM Accessbility Checker

### Description

===== DEFECT DESCRIPTION ===== 
Problem 1: The scrollable element

 with non-interactive content is not tabbable

I also noticed another one in "Deploy with IBM Cloud Schematics" in the dropdown selectors which is using NestedFilterableDropdown from pal which also has a missing role prop.

=====STEPS TO REPRODUCE =====

Press the "Review Deployment Options" button and select the radio offering "Create from CLI" which will show the CodeSnippet with the accessibility issue which Flannan found.

===== EXPECTED RESULT =====

IBM 2.1.1 Keyboard

https://www.ibm.com/able/requirements/requirements/#2_1_1

This requirement ensures that content can be navigated with a keyboard or keyboard interface. When interactive content is operated using a keyboard, it is accessible to vision impaired people who cannot use devices requiring hand-eye coordination (e.g., a mouse) and by people who use alternative keyboards or input devices that act as keyboard emulators. Keyboard emulators include speech input, sip and puff devices, on-screen keyboards, scanning software, and a variety of assistive technologies. This requirement does not forbid or discourage providing mouse input or other input methods in addition to keyboard operation.

===== SCREEN CAPTURE =====
<img width="721" alt="image" src="https://github.com/carbon-design-system/carbon/assets/110212663/edceca7c-91b4-4f00-92b6-8eb09992e8a3">

### Reproduction/example

The scrollable element <pre> with non-interactive content is not tabbable

### Steps to reproduce

1. Using Static Website: https://dev.console.test.cloud.ibm.com/catalog/architecture/static-website-7fdfef2c-cf6c-4513-aed3-295cfb33623f-global?catalog_query=aHR0cHM6Ly9kZXYuY29uc29sZS50ZXN0LmNsb3VkLmlibS5jb20vY2F0YWxvZz9zZWFyY2g9c3RhdGljI3NlYXJjaF9yZXN1bHRz

2. Press the "Review Deployment Options" button and select the radio offering "Create from CLI" which will show the CodeSnippet with the accessibility issue which Flannan found.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [ ] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by tay1orjones** [comment]
@zhanttbj In your reproduction link I don't see a "Review Deployment Options" button to be able to select the radio offering "Create from CLI"

Could you reproduce the issue in a [stackblitz](https://stackblitz.com/github/carbon-design-system/sandboxes/tree/main/react/latest/react-18-vite/?file=src%2Findex.scss,src%2FApp.jsx&preset=node=) instead, and post the link back here?

**Comment by zhanttbj** [comment]
> @zhanttbj In your reproduction link I don't see a "Review Deployment Options" button to be able to select the radio offering "Create from CLI"
> 
> Could you reproduce the issue in a [stackblitz](https://stackblitz.com/github/carbon-design-system/sandboxes/tree/main/react/latest/react-18-vite/?file=src%2Findex.scss,src%2FApp.jsx&preset=node=) instead, and post the link back here?

Pls login first, then you can see the "Review Deployment Options" button 
<img width="1184" alt="image" src="https://github.com/carbon-design-system/carbon/assets/110212663/37ba6742-0ada-44d5-99e8-aa0e65b72ca4">

**Comment by tw15egan** [comment]
@zhanttbj what version are you on? I am unable to reproduce the issue in the latest version of both v10 and v11

v10: https://v7-react.carbondesignsystem.com/?path=/story/components-codesnippet--singleline

v11: https://react.carbondesignsystem.com/?path=/story/components-codesnippet--singleline

I am able to tab to the code snippet and navigate to the right with the arrow key. Please ensure you are using the "single line" variant.

**Comment by zhanttbj** [comment]
@tw15egan here is the comments from dev, could you pls check and fix this issue if need? thanks

I was able to apply the value change to type='single' and have the issue removed for the CodeSnippet inside the Create from CLI section. But the same issue persists for the type='multi' as well. Which is showing up for the same component under Work with code. I cannot use the single variant type for the Work with code one, since it needs to show more content as multiple lines of text.

The carbon design team might need to change something for that as well since the same <pre> tag is present in there.

offering link: https://test.cloud.ibm.com/catalog/ab9dc532-d65f-452e-b509-a08addf9bcf0/architecture/custom-example-6f4d10a7-853e-458c-92e7-b090ec0a6aa4

**Comment by tw15egan** [comment]
Ahh I see, I have a simple fix that should resolve the issue 👍🏻

**Comment by zhanttbj** [comment]
@tw15egan could we have this fix also in V10? as now our services are based on carbon V10, thanks

**Comment by zhanttbj** [comment]
@tw15egan sorry, any progress on this issue? this is really block us Accessibility process, could you help on this as soon as possible, thanks in advance,

**Comment by tw15egan** [comment]
Included as part of [v10.59.16](https://github.com/carbon-design-system/carbon/releases/tag/v10.59.16)

**Comment by zhanttbj** [comment]
@tw15egan Here is the latest comments from dev, could you help to check and response? thanks

 I tried implementing the latest version 10.59.16 but it doesn't seem to have been published to the [artifactory](https://na.artifactory.swg-devops.com/ui/packages/npm:%2F%2Fcarbon-components?name=carbon-components&type=packages). So I can't download the latest changes.
<img width="1542" alt="image" src="https://github.com/carbon-design-system/carbon/assets/110212663/4830c9ac-78e0-4a70-893a-a6f628c1ef2b">

I even tried to find the correct posted version for v10 but this is what I found as the result.
As you can see the latest downloadable version is 10.58.10

Can you please ask the carbon team to investigate.

**Comment by tw15egan** [comment]
@zhanttbj the latest version for the styles (`carbon-components`) is [10.58.12](https://www.npmjs.com/package/carbon-components/v/10.58.12)

The latest version of the react package is [7.59.16](https://www.npmjs.com/package/carbon-components-react/v/7.59.16)

Please have them update to these versions.
